lib/assert.js

```diff
@@ -416,7 +416,9 @@ function _throws(shouldThrow, block, expected, message) {
       userProvidedMessage &&
       expectedException(actual, expected)) ||
       isUnexpectedException) {
-    fail(actual, expected, 'Got unwanted exception' + message);
+    fail(actual, expected,
+         'Got unwanted exception' +
+         message + '\n' + actual.message);
```

Comment:
needs a 4th argument `caller` should be either `assert.throws` or `assert.doesNotThrow`